src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -279,86 +462,39 @@ public void ProcessPushNotification(PushNotification pushNotification, TimeSpan?
                     $"{nameof(pushNotification)}.{nameof(pushNotification.ResourceUri)}");
             }
 
-            _client.UpdateSyncToken(pushNotification.SyncToken);
-
-            SetDirty(maxDelay);
+            if (_configClientManager.UpdateSyncToken(pushNotification.ResourceUri, pushNotification.SyncToken))
+            {
+                SetDirty(maxDelay);
+            }
+            else
+            {
+                _logger.LogWarning($"Ignoring the push notification received for endpoint '{pushNotification.ResourceUri}' that is not present in the input list of endpoints.");

```

Comment:
> in the input list of endpoints

The provider may not be using a list of endpoints. It may just be a connection string.

In that case, could use "Ignoring the push notification received for the unregistered endpoint '{pushNotification.ResourceUri}'"